CHANGELOG.md

```diff
@@ -31,6 +31,8 @@
 
 - [#3166](https://github.com/ChainSafe/forest/issues/3166): Add
   `forest-cli archive info` command for inspecting archives.
+- [#3159](https://github.com/ChainSafe/forest/issues/3159): Add
+  `forest-cli archive snapshot --epoch=X` command for exporting archives.
```

Comment:
Also when i try running with `forest-cli archive export --e <epoch> <input path>`:

```
➜  forest-cli archive export -e 1000 ~/.local/share/forest/snapshots/calibnet/forest_snapshot_calibnet_2023-05-03_height_526057.car 
2023-07-12T14:32:32.031317Z  INFO forest_filecoin::cli::subcommands::archive_cmd: indexing a car-backed store using snapshot: /root/.local/share/forest/snapshots/calibnet/forest_snapshot_calibnet_2023-05-03_height_526057.car
2023-07-12T14:32:35.569796Z  INFO forest_filecoin::genesis: Initialized genesis: BlockHeader: Cid(bafy2bzacecnamqgqmifpluoeldx7zzglxcljo6oja4vrmtj7432rphldpdmm2)    
2023-07-12T14:32:35.570366Z  INFO forest_filecoin::cli::subcommands::archive_cmd: looking up a tipset by epoch: 1000
Scanning blockchain 525057 / 525057 [==========================================================================================================] 100.00 % 31868.53/s 2023-07-12T14:32:52.046817Z  INFO forest_filecoin::cli::subcommands::archive_cmd: exporting snapshot at location: ./forest_snapshot_mainnet_2023-07-12_height_1000.car.zst
2023-07-12T14:32:52.046894Z  INFO forest_filecoin::chain::store::chain_store: chain export started    
Error: Cid bafy2bzaced24dlgyjj7rzfu46eog5dlqt5kpsfeom37gjkpaaetw7ludcttve not found in blockstore
```

i get `Error: Cid bafy2bzaced24dlgyjj7rzfu46eog5dlqt5kpsfeom37gjkpaaetw7ludcttve not found in blockstore` as the last line. is that expected?

Reply:
> Also when i try running with `forest-cli archive export --e <epoch> <input path>`:
> 
> ```
> ➜  forest-cli archive export -e 1000 ~/.local/share/forest/snapshots/calibnet/forest_snapshot_calibnet_2023-05-03_height_526057.car 
> 2023-07-12T14:32:32.031317Z  INFO forest_filecoin::cli::subcommands::archive_cmd: indexing a car-backed store using snapshot: /root/.local/share/forest/snapshots/calibnet/forest_snapshot_calibnet_2023-05-03_height_526057.car
> 2023-07-12T14:32:35.569796Z  INFO forest_filecoin::genesis: Initialized genesis: BlockHeader: Cid(bafy2bzacecnamqgqmifpluoeldx7zzglxcljo6oja4vrmtj7432rphldpdmm2)    
> 2023-07-12T14:32:35.570366Z  INFO forest_filecoin::cli::subcommands::archive_cmd: looking up a tipset by epoch: 1000
> Scanning blockchain 525057 / 525057 [==========================================================================================================] 100.00 % 31868.53/s 2023-07-12T14:32:52.046817Z  INFO forest_filecoin::cli::subcommands::archive_cmd: exporting snapshot at location: ./forest_snapshot_mainnet_2023-07-12_height_1000.car.zst
> 2023-07-12T14:32:52.046894Z  INFO forest_filecoin::chain::store::chain_store: chain export started    
> Error: Cid bafy2bzaced24dlgyjj7rzfu46eog5dlqt5kpsfeom37gjkpaaetw7ludcttve not found in blockstore
> ```
> 
> i get `Error: Cid bafy2bzaced24dlgyjj7rzfu46eog5dlqt5kpsfeom37gjkpaaetw7ludcttve not found in blockstore` as the last line. is that expected?

That's expected. Would be nice with better error messages, though.

Reply:
> > Also when i try running with `forest-cli archive export --e <epoch> <input path>`:
> > ```
> > ➜  forest-cli archive export -e 1000 ~/.local/share/forest/snapshots/calibnet/forest_snapshot_calibnet_2023-05-03_height_526057.car 
> > 2023-07-12T14:32:32.031317Z  INFO forest_filecoin::cli::subcommands::archive_cmd: indexing a car-backed store using snapshot: /root/.local/share/forest/snapshots/calibnet/forest_snapshot_calibnet_2023-05-03_height_526057.car
> > 2023-07-12T14:32:35.569796Z  INFO forest_filecoin::genesis: Initialized genesis: BlockHeader: Cid(bafy2bzacecnamqgqmifpluoeldx7zzglxcljo6oja4vrmtj7432rphldpdmm2)    
> > 2023-07-12T14:32:35.570366Z  INFO forest_filecoin::cli::subcommands::archive_cmd: looking up a tipset by epoch: 1000
> > Scanning blockchain 525057 / 525057 [==========================================================================================================] 100.00 % 31868.53/s 2023-07-12T14:32:52.046817Z  INFO forest_filecoin::cli::subcommands::archive_cmd: exporting snapshot at location: ./forest_snapshot_mainnet_2023-07-12_height_1000.car.zst
> > 2023-07-12T14:32:52.046894Z  INFO forest_filecoin::chain::store::chain_store: chain export started    
> > Error: Cid bafy2bzaced24dlgyjj7rzfu46eog5dlqt5kpsfeom37gjkpaaetw7ludcttve not found in blockstore
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > i get `Error: Cid bafy2bzaced24dlgyjj7rzfu46eog5dlqt5kpsfeom37gjkpaaetw7ludcttve not found in blockstore` as the last line. is that expected?
> 
> That's expected. Would be nice with better error messages, though.

I guess let's create a follow-up for this. Should be resolved by streamed snapshot walker I guess.

Reply:
The stream should give you the same error. But we can print it out in a prettier way.

Reply:
> Shouldn't it be: `forest-cli archive export --e <epoch> <input path>` command for exporting archives. ?
> 
> Tried running the former:
> 
> ```
> ➜  calibnet forest-cli archive snapshot --epoch=1000
> error: unrecognized subcommand 'snapshot'
> 
> Usage: forest-cli archive <COMMAND>
> 
> For more information, try '--help'.
> ```
> 
> but it says unrecognised sub command

Fixed the issue and Changelog.

Reply:
> The stream should give you the same error. But we can print it out in a prettier way.

Right, let me think of something.

Reply:
> The stream should give you the same error. But we can print it out in a prettier way.

Actually, I think this still belongs to `export` function and not the `archive export`. Because the former returns several different errors.

What I would do is wrap this into something like: "potentially incomplete snapshot: {err}".